test/extended/oauth/htpasswd.go

```diff
@@ -0,0 +1,69 @@
+package oauth
+
+import (
+	"path/filepath"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[Serial][Suite:openshift/oauth/serial] Htpasswd identity provider", func() {
+	defer g.GinkgoRecover()
+	var (
+		oc                    = exutil.NewCLI("htpasswd-configure", exutil.KubeConfigPath())
+		testDataBaseDir       = exutil.FixturePath("testdata", "oauth_idp")
+		oauthHtpasswdFixture  = filepath.Join(testDataBaseDir, "oauth-htpasswd.yaml")
+		htpasswdSecretFixture = filepath.Join(testDataBaseDir, "htpasswd-secret.yaml")
+		htpasswdSecretName    = "htpasswd"
+		oauthNoIdpFixture     = filepath.Join(testDataBaseDir, "oauth-noidp.yaml")
+	)
+	g.AfterEach(func() {
```

Comment:
I may not quite have the flow, but I expected a `BeforeEach` that saves the state of thing you're modifying and an after each that restores it.  This appears to be restoring a hardcoded thing maybe?